tests/Blog/BlogPostTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Tests\Blog;
+
+use DateTimeImmutable;
+use Doctrine\Website\Blog\BlogPost;
+use PHPUnit\Framework\TestCase;
+
+class BlogPostTest extends TestCase
+{
+    /** @var string */
+    private $url;
+
+    /** @var string */
+    private $title;
+
+    /** @var string */
+    private $authorName;
+
+    /** @var string */
+    private $authorEmail;
+
+    /** @var string */
+    private $contents;
+
+    /** @var DateTimeImmutable */
+    private $date;
+
+    /** @var BlogPost */
+    private $blogPost;
+
+    public function testGetUrl() : void
+    {
+        self::assertSame('http://lcl.doctrine-project.org', $this->blogPost->getUrl());
+    }
+
+    public function testGetTitle() : void
+    {
+        self::assertSame('Test Blog Post', $this->blogPost->getTitle());
+    }
+
+    public function testGetAuthorName() : void
+    {
+        self::assertSame('Jonathan H. Wage', $this->blogPost->getAuthorName());
+    }
+
+    public function testGetAuthorEmail() : void
+    {
+        self::assertSame('jonwage@gmail.com', $this->blogPost->getAuthorEmail());
+    }
+
+    public function testGetContents() : void
+    {
+        self::assertSame('Test Content', $this->blogPost->getContents());
+    }
+
+    public function testGetDate() : void
+    {
+        self::assertSame($this->date, $this->blogPost->getDate());
```

Comment:
It would fail because of the type hints, no? or I am not understanding what you mean.

Reply:
When someone is changing of types from `DateTimeImmutable` to `DateTime` in a PR for this class, that this test would fail.
But I guess that's what reviews are for.